File: MCMS/MCMS/ViewController.m
//
//  ViewController.m
//  MCMS
//
//  Created by I-Horng Huang on 29/07/2014.
//  Copyright (c) 2014 Ren. All rights reserved.
//

#import "ViewController.h"
#import "MagicalCreature.h"
#import "CreatureViewController.h"

@interface ViewController () <UITableViewDataSource, UITableViewDelegate, UITextFieldDelegate>
@property (weak, nonatomic) IBOutlet UITextField *textField;
@property (weak, nonatomic) IBOutlet UITableView *tableView;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    MagicalCreature *creature1 = [[MagicalCreature alloc]initWithName:@"creature1" image:[UIImage imageNamed:@"one"] andDescription:@"des1"];
    MagicalCreature *creature2 = [[MagicalCreature alloc]initWithName:@"creature2" image:[UIImage imageNamed:@"two"] andDescription:@"des2"];
    MagicalCreature *creature3 = [[MagicalCreature alloc]initWithName:@"creature3" image:[UIImage imageNamed:@"three"] andDescription:@"des3"];

    self.creatures = [NSMutableArray arrayWithObjects:creature1, creature2, creature3, nil];
}

- (void)viewWillAppear:(BOOL)animated
{
    [self.tableView reloadData];
}

#pragma mark - action

- (IBAction)addButtonPressed:(id)sender {

    MagicalCreature *creature = [MagicalCreature new];
    creature.name = self.textField.text;
    [self.creatures addObject:creature];

    [self.tableView reloadData];
    [self.textField resignFirstResponder];
    self.textField.text = @"";
}

#pragma mark - segue
-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{

    if ([segue.identifier isEqualToString:@"ShowCreatureSegue"]) {
        NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
        MagicalCreature *creature = [self.creatures objectAtIndex:indexPath.row];
        CreatureViewController *vc = segue.destinationViewController;
        vc.creature = creature;
    }
}


#pragma mark - delegation

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.creatures.count;
}
-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cellID"];
    MagicalCreature *creature = [self.creatures objectAtIndex:indexPath.row];
    cell.textLabel.text = creature.name;
    cell.detailTextLabel.text = creature.description;
    return cell;
}



@end

File: MeetMeUp/MeetMeUp/ViewController.swift
//
//  ViewController.swift
//  MeetMeUp
//
//  Created by I-Horng Huang on 04/08/2014.
//  Copyright (c) 2014 Ren. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    @IBOutlet weak var myTableVIew: UITableView!

    var eventArray: NSArray = []

    override func viewDidLoad() {
        super.viewDidLoad()

        var url:NSURL = NSURL(string:"https://api.meetup.com/2/open_events.json?zip=60604&text=mobile&time=,1w&key=7e724634ce575b28182845f35447")
        var request: NSURLRequest = NSURLRequest(URL: url)
        NSURLConnection.sendAsynchronousRequest(request, queue: NSOperationQueue()) { response, data, error in
            let dict = NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions(), error: nil) as NSDictionary!

            self.eventArray = dict["results"] as NSArray
            self.myTableVIew.reloadData()
        }
    }

    func tableView(tableView: UITableView!, cellForRowAtIndexPath indexPath: NSIndexPath!) -> UITableViewCell! {

            var dic:NSDictionary = self.eventArray[indexPath.row] as NSDictionary
            var cell:UITableViewCell = tableView.dequeueReusableCellWithIdentifier("cellID") as UITableViewCell
            cell.textLabel.text = dic["name"] as String
            cell.detailTextLabel.text = dic["id"] as String
            return cell
    }

    func tableView(tableView: UITableView!, numberOfRowsInSection section: Int) -> Int {
        return self.eventArray.count
    }

    override func prepareForSegue(segue: UIStoryboardSegue!, sender: AnyObject!) {
        if (segue.identifier == "segue") {
            // pass data to next view
            let indexPath = self.myTableVIew.indexPathForSelectedRow()
            let dvc = segue.destinationViewController as DetailViewController
            let dictToPass = self.eventArray[indexPath.row] as NSDictionary
            dvc.detailVcDic = dictToPass
            //passing array of dictionaries
        }
    }
}
















